rancid-tomatillos/src/App.js

```diff
@@ -1,30 +1,54 @@
 import React, {Component} from "react";
+import {BrowserRouter as Router, Switch, Route, Link} from "react-router-dom";
 import MainView from "./components/MainView";
-import CardsContainer from "./components/CardsContainer";
+import Header from "./components/Header";
+import Login from "./components/Login";
+import ShowPage from "./components/ShowPage";
 
 import "./App.css";
 
 class App extends Component {
     constructor() {
         super();
+        this.setLoggedIn = this.setLoggedIn.bind(this);
         this.state = {
-            movies: null
+            movies: null,
+            userId: 0,
+            loggedIn: false,
         };
     }
-    componentDidMount() {
-        return fetch("https://rancid-tomatillos.herokuapp.com/api/v2/movies")
-            .then((response) => response.json())
-            .then((data) => {
-                this.setState({movies: data.movies});
-            });
+    async componentDidMount() {
+        let promise = await fetch(
+            "https://rancid-tomatillos.herokuapp.com/api/v2/movies"
+        );
+        let result = await promise.json();
+        this.setState({movies: result.movies});
+    }
+    setLoggedIn() {
+        this.setState({loggedIn: true});
     }
 
     render() {
         return (
-            <div>
-                <MainView />
-                <CardsContainer allMovies={this.state.movies} />
-            </div>
+            <Router>
+                <Header />
+                <div className="page-container">
+                    <Switch>
+                        <Route path="/users">
+                            <ShowPage />
+                        </Route>
+                        <Route
+                            path="/login"
+                            render={(props) => (
+                                <Login {...props} isAuthed={this.setLoggedIn} />
+                            )}
```

Comment:
Finally got this to pass in both Route props and our callback functions!!! wohoo!!

Reply:
Was this a change you made afterwards? I think I remember it being destructured props and isAuthed may have been hard-coded where we left off

Reply:
Yeah, this was after we left off, sorry. 
Adding props to the callback function allows for the Route props to also get passed in. 
isAuthed is just the name of the prop holding the {this.setLoggedIn} callback. 
We got to almost this exact spot before we got off the call, we just needed to move the syntax around slightly.